qemu/tests/virtio_chardev_trace.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    virtio-trace support testing:
+    1) Make FIFO per CPU in a host.
+    2) reboot guest with virtio-serial device, control path and data path per CPU.
+    3) Download trace agent and compile it.
+    4) Enable ftrace in the guest.
+    5) Run trace agent in the guest.
+    6) Open FIFO in a host.
+    7) Start to read trace data by ordering from a host
+    8) Stop to read trace data by ordering from a host
+    9) repeat 7) and 8) with different CPU
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def add_serial_device(params, device, bus, nr, chardev, _id, name):
+        params[
+            "extra_params"
+        ] += " -device {},bus={},nr={},chardev={},id={},name={} ".format(
+            device, bus, nr, chardev, _id, name
+        )
+
+    def add_chardev(params, backend, _id, path):
+        params["extra_params"] += " -chardev {},id={},path={} ".format(
+            backend, _id, path
+        )
```

Comment:
```
for i in range(1, *smp* + 1):
    serials += 'vs%s' %i
```
Then define serial_type for those serials, after start vm, framework will add those device automatically.
Do not add device with extra_params string.